src/App/Controller/ThemeController.php

```diff
@@ -134,37 +131,57 @@ public function editAction(Request $request, $slug)
 
         // in this array are the id of the theme considered, and the ids of all its descendants
         $descendantsId = $repository->getChildrenId($theme, false, null, 'ASC', true);
-        $form          = $this->createForm(new EditThemeType($descendantsId), $theme);
+        $form = $this->createForm(new MoveThemeType($descendantsId), $theme);
         $form->handleRequest($request);
 
-        if ($form->isValid()) {
-            $theme->setParent($form->get('parent')->getData());
+        if ($form->isSubmitted() && $form->isValid()) {
             $em = $this->getDoctrine()->getManager();
             $em->persist($theme);
             $em->flush();
 
             return $this->redirect($this->generateUrl('theme_show', [
-                    'slug' => $theme->getSlug(),
+                'slug' => $theme->getSlug(),
             ]));
         }
 
-        return $this->render('App:Theme:edit.html.twig', [
-            'form' => $form->createView(),
+        return $this->render('App:Theme:move_theme.html.twig', [
+            'form'   => $form->createView(),
+            'theme'  => $theme,
         ]);
     }
 
-    public function showDiscussionsAction($slug)
+    /**
+     * @Security("has_role('ROLE_ADMIN')")
+     */
+    public function deleteAction(Request $request, $slug)
     {
-        $theme = $this->getDoctrine()->getRepository('App:Theme')->findOneBySlug($slug);
+        $repository = $this->getDoctrine()->getRepository('App:Theme');
+        $theme      = $repository->findOneBySlug($slug);
 
         if (null === $theme) {
             throw $this->createNotFoundException();
         }
 
-        $discussions = $this->getDoctrine()->getRepository('App:Discussion')->findByTheme($theme);
+        if (false === $theme->isEmpty()) {
+            $this->get('session')->getFlashBag()->add('info', 'A theme cannot be deleted if it contains any proposal');
 
-        return $this->render('App:Theme:discussions.html.twig', [
-            'discussions' => $discussions,
-        ]);
+            return $this->redirect($this->generateUrl('theme_show', [
+                'slug' => $theme->getSlug(),
+            ]));
+        }
+
+        $parent = $theme->getParent();
+        $repository->removeFromTree($theme);
+        $this->getDoctrine()->getManager()->clear(); // clear cached nodes
+        // it will remove this node from tree and reparent all children
+        $this->get('session')->getFlashBag()->add('info', 'The theme entitled '.$theme->getTitle().' has been deleted');
```

Comment:
You can directly use the method `addFlashBag()`

Reply:
Why is this flashbag in english if everything else is in french?

Reply:
Done